dataGeneration/generateData.js

```diff
@@ -0,0 +1,143 @@
+const faker = require('faker');
+const fs = require('fs');
+// // \l - shows all databases
+// // \d - shows all tables
+
+
+// generatePaymentMethods(); //10s
+// generateUserTransData(); //3min
+generatePurchasedProds(); // 7 - 8 mins
+
+function getRandomArbitrary(min, max) {
+  return Math.floor(Math.random() * (max - min) + min);
+}
+
+//two payment Methods for each user
+function generatePaymentMethods() {
+ var first = true;
+  var writeStream = fs.createWriteStream('./dataFiles/PaymentMethods.csv', {'flags': 'a', 'encoding': null});
+  var typeOfPayment = ['Visa', 'Master Card', 'Discover', 'UCB', 'Diners Club', 'UnionPay', 'American Express']
+  writeStream.once('open', (fd) => {
+    if (first) {
+      first = false;
+      writeStream.write('id; userId\n');
+    }
+    count = 1;
+    second = false;
+    //4000000
+    for (var i = 1; i <= 4000000; i++) {
+      var paymentId = i;
+      // var lastFourDigits = faker.helpers.replaceSymbolWithNumber("####");
+      // var type = typeOfPayment[Math.floor(Math.random() * typeOfPayment.length)];
+      var userId = count;
+
+      if (second) {
+        second = false;
+        count += 1;
+      } else {
+        second = true;
+      }
+      // REMEMBER PAYMENTID IS NOT NECCESSARILY  1-..., IT COULD BE RANDOM UNIQUE IDS
+      writeStream.write(`${paymentId}; ${userId}\n`);
+    }
+    writeStream.end();
+    console.log('done');
+  });
+}
+
+//a user makes one transactions
+//first step is to write 1 line of data that resembles my schema to a csv file.
+function generateUserTransData() {
+  var first = true;
+  var writeStream = fs.createWriteStream('./dataFiles/UserTrans.csv', {'flags': 'a', 'encoding': null});
+
+  writeStream.once('open', (fd) => {
+    if (first) {
+      first = false;
+      writeStream.write('date; userId; paymentMethodId; status; fullName; addressLine1; addressLine2; city; state; zip; country; phone; grandTotal\n');
+    } 
+    var statusArr = ['pending', 'completed', 'failed'];
+    count =- 1;
+    //2000000
+    for (var i = 1; i <= 2000000; i++) {
+      if (i % 100000 === 0) {
+        console.log(i);
+      }
+      var userTransId = i; //number from 1 - 2M
+      var date = faker.date.past().toISOString();
+
+      var userId = i; //could be random uui,
+      //insert into payment methods table userId 2 times id, id+1
+      count = count + 2;
+      var paymentMethodId = count; //each user has different paymentIds
+
+      var status = statusArr[Math.floor(Math.random() * 3)];
+      var fullName = faker.name.findName();
+      var addressLine1 = faker.fake("{{address.streetAddress}}");
```

Comment:
You may wish to write the faker data in the same format. For instance faker.name.findName() and faker.address.streetAddress() or the other way around.